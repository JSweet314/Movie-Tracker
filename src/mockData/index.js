export const mockMovies = [{
  "vote_count": 56,
  "id": 437670,
  "video": false,
  "vote_average": 6.2,
  "title": "Suck Me Shakespeer 3",
  "popularity": 28.024527,
  "poster_path": "/cypnifmPJ5JbTjzpZI6MwJdvP7.jpg",
  "original_language": "de",
  "original_title": "Fack ju GÃ¶hte 3",
  "genre_ids": [35],
  "backdrop_path": "/a3Ipy3B4APTPx1ISkls95XSs5fU.jpg",
  "adult": false,
  "overview": "A comedy that follows an con who lands a position at a",
  "release_date": "2017-10-26"
}];

export const mockMovie = [{
  "movieId": 437670,
  "voteAverage": 6.2,
  "title": "Suck Me Shakespeer 3",
  "posterPath": "/cypnifmPJ5JbTjzpZI6MwJdvP7.jpg",
  "overview": "A comedy that follows an con who lands a position at a",
  "releaseDate": "2017-10-26"
}];

