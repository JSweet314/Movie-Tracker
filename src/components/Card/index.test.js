import { shallow } from 'enzyme';
import {Card} from './index.js';
import React from 'react';
import * as mockData from '../../mockData';

describe('Card', () => {
  let wrapper;
  const movie = mockData.mockMovie[0];
  const mockAddFavorite = jest.fn();
  beforeEach(() => {
    wrapper = shallow(
      <Card 
        information={movie} 
        addFavorite={mockAddFavorite} 
        logStatus={true}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call addFavorites with the right params on click', () => {
    wrapper.find('button').simulate('click');
    expect(mockAddFavorite).toHaveBeenCalledWith(movie);

  });
});
