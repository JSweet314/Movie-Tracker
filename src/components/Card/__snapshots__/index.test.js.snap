// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    handleFavorite={[Function]}
    information={
        Object {
            "movieId": 437670,
            "overview": "A comedy that follows an con who lands a position at a",
            "posterPath": "/cypnifmPJ5JbTjzpZI6MwJdvP7.jpg",
            "releaseDate": "2017-10-26",
            "title": "Suck Me Shakespeer 3",
            "voteAverage": 6.2,
          }
    }
    logStatus={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="movie-title"
>
          Suck Me Shakespeer 3
</h3>,
        <button
          className="favorite-btn"
          onClick={[Function]}
>
          Favorite
</button>,
        <p
          className="info"
>
          Released: 
          2017-10-26
</p>,
        <p
          className="info"
>
          Rating: 
          6.2
</p>,
        <p
          className="movie-overview"
>
          A comedy that follows an con who lands a position at a
</p>,
      ],
      "className": "movie-card",
      "style": Object {
        "backgroundImage": "url(https://image.tmdb.org/t/p/w500//cypnifmPJ5JbTjzpZI6MwJdvP7.jpg)",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Suck Me Shakespeer 3",
          "className": "movie-title",
        },
        "ref": null,
        "rendered": "Suck Me Shakespeer 3",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Favorite",
          "className": "favorite-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Favorite",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Released: ",
            "2017-10-26",
          ],
          "className": "info",
        },
        "ref": null,
        "rendered": Array [
          "Released: ",
          "2017-10-26",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Rating: ",
            6.2,
          ],
          "className": "info",
        },
        "ref": null,
        "rendered": Array [
          "Rating: ",
          6.2,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "A comedy that follows an con who lands a position at a",
          "className": "movie-overview",
        },
        "ref": null,
        "rendered": "A comedy that follows an con who lands a position at a",
        "type": "p",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="movie-title"
>
            Suck Me Shakespeer 3
</h3>,
          <button
            className="favorite-btn"
            onClick={[Function]}
>
            Favorite
</button>,
          <p
            className="info"
>
            Released: 
            2017-10-26
</p>,
          <p
            className="info"
>
            Rating: 
            6.2
</p>,
          <p
            className="movie-overview"
>
            A comedy that follows an con who lands a position at a
</p>,
        ],
        "className": "movie-card",
        "style": Object {
          "backgroundImage": "url(https://image.tmdb.org/t/p/w500//cypnifmPJ5JbTjzpZI6MwJdvP7.jpg)",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Suck Me Shakespeer 3",
            "className": "movie-title",
          },
          "ref": null,
          "rendered": "Suck Me Shakespeer 3",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorite",
            "className": "favorite-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Favorite",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Released: ",
              "2017-10-26",
            ],
            "className": "info",
          },
          "ref": null,
          "rendered": Array [
            "Released: ",
            "2017-10-26",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Rating: ",
              6.2,
            ],
            "className": "info",
          },
          "ref": null,
          "rendered": Array [
            "Rating: ",
            6.2,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "A comedy that follows an con who lands a position at a",
            "className": "movie-overview",
          },
          "ref": null,
          "rendered": "A comedy that follows an con who lands a position at a",
          "type": "p",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
